server/src/main/java/com/flipkart/grayskull/controllers/SecretController.java

```diff
@@ -0,0 +1,101 @@
+package com.flipkart.grayskull.controllers;
+
+import com.flipkart.grayskull.models.dto.request.CreateSecretRequest;
+import com.flipkart.grayskull.models.dto.request.UpgradeSecretDataRequest;
+import com.flipkart.grayskull.models.dto.response.*;
+import com.flipkart.grayskull.spi.models.enums.LifecycleState;
+import com.flipkart.grayskull.service.interfaces.SecretService;
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/v1/project/{projectId}/secrets")
```

Comment:
plurals. `.../projects/...`

Reply:
This just means that it was a miss in the contract proposal review. Please fix this.